### PR TITLE
Add withdrawal reasons for placement requests and placement applications

### PR DESCRIPTION
**Pull request by davidatkinsuk**
https://dsdmoj.atlassian.net/browse/APS-246

Also includes a logic change, allowing applicant to withdraw their own placement requests (previously this could only be done by a manager)

### CODE REVIEW

File: src/main/resources/static/_shared.yml [hunk 1]
```diff
@@ -505,6 +505,16 @@ components:
...
         - arrivalDate
         - departureDate
+    WithdrawPlacementRequest:
+      type: object
+      properties:
+        reason:
+          type: string
+          enum:
+            - DuplicatePlacementRequest
+            - AlternativeProvisionIdentified
```
Comment (pezholio) [inline src/main/resources/static/_shared.yml:515]: Could we make this enum a named object? That way it'll be easier to use in the frontend

File: src/main/resources/static/_shared.yml [hunk 2]
```diff
@@ -3392,6 +3392,16 @@ components:
...
         - release_following_decision
         - additional_placement
+    WithdrawPlacementApplication:
+      type: object
+      properties:
+        reason:
+          type: string
+          enum:
+            - DuplicatePlacementRequest
+            - AlternativeProvisionIdentified
```
Comment (pezholio) [inline src/main/resources/static/_shared.yml:3402]: Same here vis a vis the named object for the enum. Could potentially share it with the `WithdrawPlacementRequest` object too

**Review by pezholio — APPROVED** [review]
Looks good. Minor suggestion about the API objects

File: src/main/resources/static/_shared.yml (hunk 1 above)
Comment (davidatkinsuk) [reply src/main/resources/static/_shared.yml]: Yep, i'll remember to do that for new enums going forward

File: src/main/resources/static/_shared.yml (hunk 2 above)
Comment (davidatkinsuk) [reply src/main/resources/static/_shared.yml]: Good idea! have done this

**Pull request merged**